### DISCUSSION TITLE
How do I reset the `window` or `globalThis` for every new test?

### DISCUSSION THREAD

**Discussion by Blankeos** [capricorn86/happy-dom#1643] [Q&A]
I can't seem to just use this because it does not work.

```ts
describe("a", () => {
   it("b", () => {
      const window = new Window();
      const document = window.document;
      // add globalThis.test here.

     expect(window["test"]).not.toBe(undefined); // does not work. 
   });

   it ("c", () => {
      const window = new Window();
      const document = window.document; 
   });
})
```

What works is if I use the GlobalRegistrator. The same code would work. BUt the `it("c")` will inherit the new global variable `"test"` from the previous **it**

**Comment by capricorn86** [comment]
> I can't seem to just use this because it does not work.
> 
> ```ts
> describe("a", () => {
>    it("b", () => {
>       const window = new Window();
>       const document = window.document;
>       // add globalThis.test here.
> 
>      expect(window["test"]).not.toBe(undefined); // does not work. 
>    });
> 
>    it ("c", () => {
>       const window = new Window();
>       const document = window.document; 
>    });
> })
> ```
> 
> What works is if I use the GlobalRegistrator. The same code would work. BUt the `it("c")` will inherit the new global variable `"test"` from the previous **it**

The easiest way is to just clean up after the test.

One way that will make your code execute within the Window context is to run it in a script inside the Window:
```
 const window = new Window();
 window.document.write(`<script>window.test = true;</script>`);
 ```
 
 What testing tool are you using? Vitest has test isolation that might solve it for you.
